### PR TITLE
Added release notes for Docker Desktop 3.0.4

### PR DESCRIPTION
**Pull request by mat007**
### Proposed changes

Added release notes for Docker Desktop 3.0.4

### CODE REVIEW

**Review by djs55 — DISMISSED** [review]
LGTM

File: docker-for-mac/release-notes.md [hunk 1]
```diff
@@ -16,10 +16,23 @@ This page contains information about the new features, improvements, known issue
...
+
+> [Download](https://desktop.docker.com/mac/stable/Docker.dmg)
+
+### Upgrades
+
+- [Docker Engine 20.10.2](https://docs.docker.com/engine/release-notes/#20102)
+
+### Bug fixes and minor changes
+
+- Fixed a failure with mounting overlapping volumes. See [docker/for-mac#5157](https://github.com/docker/for-mac/issues/5157).
```
Comment (stephen-turner) [inline docker-for-mac/release-notes.md:30]: This is not correct. 5157 was fixed in 3.0.3: it was just done by breaking 4957 and 5124. So the fix in 3.0.4 relative to 3.0.3 is that the latter two issues are now fixed (and 5157 isn't broken again, but we don't need to write that).

See the 3.0.3 release notes for more context.

File: docker-for-windows/release-notes.md [hunk 2]
```diff
@@ -16,6 +16,19 @@ This page contains information about the new features, improvements, known issue
...
+
+> [Download](https://desktop.docker.com/win/stable/Docker%20Desktop%20Installer.exe)
+
+### Upgrades
+
+- [Docker Engine 20.10.2](https://docs.docker.com/engine/release-notes/#20102)
+
+### Bug fixes and minor changes
+
+- Fixed an issue causing Docker Desktop to fail to start. See [docker/for-win#955](https://github.com/docker/for-win/issues/9755).
```
Comment (stephen-turner) [inline docker-for-windows/release-notes.md:30]: "Fixed an issue that could cause Docker Desktop to fail to start after upgrading to 3.0.0".

Comment (stephen-turner) [reply docker-for-windows/release-notes.md:30]: Also typo: 9755 not 955.

File: docker-for-mac/release-notes.md [hunk 3]
```diff
@@ -16,10 +16,23 @@ This page contains information about the new features, improvements, known issue
...
+
```
Comment (stephen-turner) [inline docker-for-mac/release-notes.md:31]: I think the Alpine known issue is not yet fixed on this branch so we should carry it forward in the release notes.

**Review by stephen-turner — CHANGES_REQUESTED** [review]

File: docker-for-mac/release-notes.md (hunk 1 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:30]: Right, so what do we say here? That we fixed 4957 and 5124 again?

Comment (mat007) [reply docker-for-mac/release-notes.md:30]: OK, that actually makes sense, I’m adding back
```
- Avoid timeouts during `docker-compose up` by making cache invalidation faster. Fixes [docker/for-mac#4957](https://github.com/docker/for-mac/issues/4957).
- Avoid generating a spurious filesystem DELETE event while invalidating caches. Fixes [docker/for-mac#5124](https://github.com/docker/for-mac/issues/5124).
```

**Review by stephen-turner — APPROVED** [review]
Looks good now.

**Pull request merged**